src/cacao.ts

```diff
@@ -78,6 +84,45 @@ export namespace Cacao {
 
     return cacao
   }
+
+  export function verify(cacao: Cacao): VerificationResult {
+    if (cacao.h.t === 'eip4361-eip191') {
+      return verifyEIP191Signature(cacao)
+    }
+    throw new Error('Unsupported CACAO signature type')
+  }
+
+  export function verifyEIP191Signature(cacao: Cacao): VerificationResult {
+    try {
+      if (!cacao.s) {
+        throw new Error(`CACAO does not have a signature`)
+      }
+
+      if (cacao.p.iat > Date.now() || cacao.p.nbf > Date.now()) {
+        throw new Error(`CACAO is not valid yet`)
+      }
+
+      if (cacao.p.exp < Date.now()) {
+        throw new Error(`CACAO has expired`)
+      }
+
+      const msg = SiweMessage.fromCacao(cacao)
+      const sig = cacao.s.s
+      const recoveredAddress = verifyMessage(msg.toMessage(), sig)
+      if (recoveredAddress.toLowerCase() !== cacao.p.iss.toLowerCase()) {
```

Comment:
According to the [spec](https://github.com/ChainAgnostic/CAIPs/pull/74) `cacao.p.iss` is a DID PKH, so we need to parse it using the `caip` library.

Reply:
Woops, great catch!